src/App.tsx

```diff
@@ -1,73 +1,105 @@
-import React from 'react';
 import './App.scss';
+import { useState, useCallback, useEffect, useRef, useMemo } from 'react';
+import cn from 'classnames';
+import debounce from 'lodash.debounce';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import { PeopleList } from './components/PeopleList';
+
+const DELAY = 300;
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const dropRef = useRef<HTMLDivElement>(null);
+
+  const [selected, setSelected] = useState<Person | null>(null);
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [touched, setTouched] = useState(false);
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const applyQuery = useCallback(debounce(setAppliedQuery, DELAY), []);
+
+  const normalizeQuery = appliedQuery.trim().toLowerCase();
```

Comment:
More just FYI:
It's not a big deal here, but `normalizedQuery` will be created again on every render since it's not memorized, so the same for the `filteredPeople`.
In such case, it's better to normalize it inside `filteredPeople` useMemo's callback, and set `appliedQuery` as a dependency.